### DISCUSSION TITLE
How can I read/write resources in Avalonia?

### DISCUSSION THREAD

**Discussion by kyuranger** [AvaloniaUI/Avalonia#15375] [Q&A] [answered]
I need to read/write resources file inside my program (Not implemented through a separate file).

Here is a tutorial about this: https://stackoverflow.com/questions/5599662/how-to-write-to-a-resource-file

However, System.Resources.ResXResourceReader and System.Resources.ResXResourceWriter are part of System.Windows.Forms that they cannot cross platform.

Does Avalonia have a similar function to achieve this? Thank you.

**Comment by maxkatz6** [comment, marked as answer]
It's not really a responsibility for Avalonia. 
We have AvaloniaResource and AssetsLoader though for avalonia resources specifically. https://docs.avaloniaui.net/docs/basics/user-interface/assets

For resx you can try https://github.com/farlee2121/ResXResourceReader.NetStandard

**Reply by maxkatz6** [reply]
But ResourceManager API for RESX also works without winforms:
https://learn.microsoft.com/en-us/dotnet/api/system.resources.resourcemanager?view=net-8.0

**Reply by kyuranger** [reply]
I tried to use AssetsLoader to write an AvaloniaResources like this:
```
var _embeddedAsset=AssetLoader.Open(new Uri($"avares://AvaloniaApplication1/Properties/EmbeddedAsset"));
var ss = Encoding.UTF8.GetBytes("aaaa");
_embeddedAsset.Write(ss, 0, ss.Count());
_embeddedAsset.Close();
```

However, Visual Studio reports an error:
```
System.NotSupportedException
  HResult=0x80131515
  Message=Specified method is not supported.
  Source=Avalonia.Base
  StackTrace:
   at Avalonia.Platform.Internal.SlicedStream.Write(Byte[] buffer, Int32 offset, Int32 count)
```

**Reply by maxkatz6** [reply]
These are not writeable resources. You embed them via AvaloniaResources and that's it.

**Reply by maxkatz6** [reply]
In general, you can't really write to any resources embedded in dlls nowadays. You probably should find another approach.

**Reply by kyuranger** [reply]
All right. Thank you very much.
